File: Arcadia_Mobile/Player/CompPlayer.cs
using Arcadia_Mobile.Creatures;
using Arcadia_Mobile.Map;

namespace Arcadia_Mobile.Player;

public sealed class CompPlayer : GenericPlayer
{
    private readonly List<Animal> battleTeamTemplate = new();

    public CompPlayer(string name, Room startingRoom)
        : base(name, startingRoom)
    {
    }

    public bool Defeated { get; set; }
    public IReadOnlyList<Animal> BattleTeamTemplate => battleTeamTemplate;

    public void SetBattleTeam(IEnumerable<Animal> templateAnimals)
    {
        ArgumentNullException.ThrowIfNull(templateAnimals);

        battleTeamTemplate.Clear();

        foreach (Animal animal in templateAnimals)
        {
            ArgumentNullException.ThrowIfNull(animal);
            battleTeamTemplate.Add(animal.Clone());
        }

        PrepareForBattle();
    }

    public void PrepareForBattle()
    {
        ClearAnimalInventory();

        foreach (Animal animal in battleTeamTemplate)
        {
            AddAnimal(animal.Clone());
        }
    }
}

File: arcadia_flutter/lib/player/comp_player.dart
import '../creatures/animal.dart';
import 'generic_player.dart';

class CompPlayer extends GenericPlayer {
  CompPlayer({required super.name, required super.startingRoom});

  final List<Animal> _battleTeamTemplate = [];

  bool defeated = false;

  List<Animal> get battleTeamTemplate => List.unmodifiable(_battleTeamTemplate);

  void setBattleTeam(Iterable<Animal> templateAnimals) {
    _battleTeamTemplate
      ..clear()
      ..addAll(templateAnimals.map((animal) => animal.clone()));

    prepareForBattle();
  }

  void prepareForBattle() {
    clearAnimalInventory();

    for (final animal in _battleTeamTemplate) {
      addAnimal(animal.clone());
    }
  }
}
